src/codegen/c.zig

```diff
@@ -2489,16 +2496,50 @@ fn genExports(o: *Object) !void {
 
     const mod = o.dg.module;
     const ip = &mod.intern_pool;
-    const fwd_decl_writer = o.dg.fwd_decl.writer();
-    if (mod.decl_exports.get(o.dg.decl_index.unwrap().?)) |exports| {
-        for (exports.items[1..], 1..) |@"export", i| {
-            try fwd_decl_writer.writeAll("zig_export(");
-            try o.dg.renderFunctionSignature(fwd_decl_writer, o.dg.decl_index.unwrap().?, .forward, .{ .export_index = @as(u32, @intCast(i)) });
-            try fwd_decl_writer.print(", {s}, {s});\n", .{
-                fmtStringLiteral(ip.stringToSlice(exports.items[0].opts.name), null),
-                fmtStringLiteral(ip.stringToSlice(@"export".opts.name), null),
-            });
-        }
+    const decl = o.dg.decl.?;
+    const decl_index = o.dg.decl_index.unwrap().?;
+    const tv: TypedValue = .{ .ty = decl.ty, .val = (try decl.internValue(mod)).toValue() };
+    const fwd = o.dg.fwd_decl.writer();
+
+    const exports = mod.decl_exports.get(decl_index) orelse return;
+    if (exports.items.len < 2) return;
+
+    switch (ip.indexToKey(tv.val.toIntern())) {
+        .func => {
+            const not_windows = mod.getTarget().os.tag != .windows;
```

Comment:
Checking for windows in this file is almost always incorrect, usually what you actually want to check for is msvc but that can't be done until `zig.h`, so you'll need to split `zig_weak_linkage` into two defines, just like `zig_align` and `zig_align_fn`.